.github/workflows/pull-request-main.yml

```diff
@@ -0,0 +1,127 @@
+name: Pull request (main) CI tests
+
+on:
+  push:
+    branches:
+      - main
+      # TODO "gh-actions" branch included to facilitate development
+      # we could also consider simply going myfork:main -> upstream:main for CI PRs
+      - gh-actions
```

Comment:
Should this be removed before merging?

Reply:
Good point, I think so. It looks like the `on: push` trigger works just fine, and in case there are more reviews requesting changes that I need to test, I should be able to trigger the workflows manually.